packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -28,17 +28,31 @@ const getNextCellIndexes = (code: string, indexes: GridCellIndexCoordinates) =>
     throw new Error('Material-UI: The first argument (code) should be an arrow key code.');
   }
 
-  if (code === 'ArrowLeft') {
-    return { ...indexes, colIndex: indexes.colIndex - 1 };
-  }
-  if (code === 'ArrowRight') {
-    return { ...indexes, colIndex: indexes.colIndex + 1 };
-  }
-  if (code === 'ArrowUp') {
-    return { ...indexes, rowIndex: indexes.rowIndex - 1 };
+  switch (code) {
```

Comment:
code is confusing, it's coming from `event.key` 

```suggestion
  switch (key) {
```